bot.py

```diff
@@ -709,11 +709,216 @@ def cmd_shoutout(match: Match[str]) -> Response:
     )
 
 
+class YouTubeVideo(NamedTuple):
+    video_id: str
+    title: str
+    playlist_id: str
+    playlist_position: int
+
+    def __repr__(self) -> str:
+        return (
+            f'{self.title} - '
+            f'https://www.youtube.com/watch?v={self.video_id}&list={self.playlist_id}&index={self.playlist_position}'  # noqa: E501
+        )
+
+
+async def ensure_youtube_playlist_videos_table_exists(
+    db: aiosqlite.Connection
+) -> None:
+    await db.execute(
+        'CREATE VIRTUAL TABLE IF NOT EXISTS youtube_playlist_videos using FTS5'
```

Comment:
storing these in memory via an alru_cache is going to be simpler

Reply:
Simpler yes, but not as good for text searching I would argue. I feel like trying to roll-your-own text search won't be as good as a DB's one?

Reply:
ah that makes sense, I completely glossed over the full text search part